draft-ietf-tls-ctls.md

```diff
@@ -354,15 +247,43 @@ group is listed by the code point name in {{RFC8446}}, Section 4.2.7.
 (e.g., ed25519). This implies a literal "signature_algorithms" extension
 consisting solely of this group.
 
-randomSize (integer):
+random (integer):
 : indicates that the ClientHello.Random and ServerHello.Random values
 are truncated to the given values. When the transcript is
 reconstructed, the Random is padded to the right with 0s and the
 anti-downgrade mechanism in {{RFC8446)}, Section 4.1.3 is disabled.
 IMPORTANT: Using short Random values can lead to potential
-attacks. When Random values are shorter than 8 bytes, PSK-only modes
-MUST NOT be used, and each side MUST use fresh DH ephemerals.
-The Random length MUST be less than or equal to 32 bytes.
+attacks. The Random length MUST be less than or equal to 32 bytes.
+When this element is omitted, the random values are created by hashing 
+the ephemeral public key via the hash algorithm indicated in 
+cipherSuite and truncated to 32 bytes. 
+
+mutualAuth (boolean):
+: indicates whether the client has to be prepared to transmit a 
+Certificate and a CertificateVerify message and whether the server 
+has to transmit a CertificateRequest message. 
+
+extension_order: 
+: indicates in what order extensions appear in respective messages. 
+This allows to omit sending the type. If there is only a single 
+extension to be transmitted, then the extension length field can also
+be omitted. For example, imagine that only the KeyShare extension 
+needs to be sent in the ClientHello as the only extension. Then, 
+the following structure
+
+~~~
+   28                    // Extensions.length
+   33 26                 // KeyShare
+     0024                // client_shares.length
+       001d              // KeyShareEntry.group
+       0020 a690...af948 // KeyShareEntry.key_exchange
+~~~       
+
+is compressed down to
+
+~~~
+   0020 a690...af948 // KeyShareEntry.key_exchange
+~~~
```

Comment:
This doesn't seem right unless you are pinning down the entire group.

ISTM that one could have:

- Extension order: then you just get to omit the type and length (I believe all current extensions are internally self-delimited)
- One group key shares.

And you could use one or both.

Reply:
I added a clarifying note.